### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by serentinos**

### CODE REVIEW

**Review by skinnyjazz — COMMENTED** [review]
It looks nice, but there shouldn't be any commented code.

**Review by AleksandrTsivilin — APPROVED** [review]
Good job!!

File: src/arrayMethodJoin.js [hunk 1]
```diff
@@ -4,9 +4,26 @@
...
+    }
+
+    return newStr;
   };
 }
 
+// applyCustomJoin();
+
+// ['test', 'test'].join2('--');
+
```
Comment (33Orange) [inline src/arrayMethodJoin.js]: Delete comments

File: src/arrayMethodJoin.js [hunk 2]
```diff
@@ -4,9 +4,26 @@
...
+    const lastIndex = this.length - 1;
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== undefined && this[i] !== null) {
+        newStr += this[i];
+      };
+
+      if (i !== lastIndex) {
+        newStr += `${separator}`;
+      };
```
Comment (33Orange) [inline src/arrayMethodJoin.js:18]: Remove the unnecessary semicolon after the closing brace of the if statement.

File: src/arrayMethodJoin.js [hunk 3]
```diff
@@ -4,9 +4,26 @@
...
+    }
+
+    return newStr;
   };
 }
 
+// applyCustomJoin();
```
Comment (33Orange) [inline src/arrayMethodJoin.js:25]: Remove the commented-out code or provide a meaningful comment explaining its purpose.

File: src/arrayMethodJoin.js [hunk 4]
```diff
@@ -4,9 +4,26 @@
...
+
+// ['test', 'test'].join2('--');
```
Comment (33Orange) [inline src/arrayMethodJoin.js:27]: Remove the commented-out code or provide a meaningful comment explaining its purpose.

File: src/arrayMethodJoin.js [hunk 5]
```diff
@@ -4,9 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
```
Comment (33Orange) [inline src/arrayMethodJoin.js:7]: Avoid extending the built-in Array prototype. Instead, create a new function that takes array as a parameter.

File: src/arrayMethodJoin.js [hunk 6]
```diff
@@ -4,9 +4,26 @@
...
+    let newStr = '';
+    const lastIndex = this.length - 1;
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== undefined && this[i] !== null) {
+        newStr += this[i];
+      };
```
Comment (33Orange) [inline src/arrayMethodJoin.js:14]: Remove the unnecessary semicolon after the closing brace of the if statement.

**Review by 33Orange — CHANGES_REQUESTED** [review]
Good job 🔥 

Don't forget to check the code and remove useless comments, console.log, etc.

**Review by Anna-Tkachenko — APPROVED** [review]

**Pull request closed without merging**